### PR TITLE
Adds proper MacOS bundle and icons for Windows builds

### PR DESCRIPTION
**Pull request by dethrace-labs**
- MacOS binary bundled with the SDL dependency as a `.app` , distributed inside a disk image.
- Windows binary contains an icon

Windows:
Screenshot 2024-07-03 at 1 55 46 PM

MacOS:
Screenshot 2024-07-03 at 1 56 47 PM

### CODE REVIEW

File: src/DETHRACE/CMakeLists.txt [hunk 1]
```diff
@@ -195,8 +219,16 @@ endif()
 
 if (DETHRACE_INSTALL)
     install(TARGETS dethrace
-        RUNTIME DESTINATION "."
+        BUNDLE DESTINATION . COMPONENT Runtime
+        RUNTIME DESTINATION bin COMPONENT Runtime
```
Comment (madebr) [inline src/DETHRACE/CMakeLists.txt:223]:
```suggestion
        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT Runtime
```
And do `include(GNUInstallDirs)` somewhere before this point.

Comment (madebr) [reply src/DETHRACE/CMakeLists.txt:223]: This also means the windows zip will have `dethrace.exe` in a bin directory.
So perhaps we want to configure with `-DCMAKE_INSTALL_BINDIR=.` on Windows,
or do the following before `include(GNUInstallDirs)`:
```cmake
if(WIN32)
  set(CMAKE_INSTALL_BINDIR "." CACHE PATH "Binary installation directory")
endif()
include(GNUInstallDirs)
```

Also, make sure to install the extra runtime dlls (SDL2.dll) and the pdb (SDL2.pdb) to the same directory in [the lines below](https://github.com/dethrace-labs/dethrace/blob/673244bb8c19cdd3041bfd69e0891c504f58757a/src/DETHRACE/CMakeLists.txt#L196-L212)

Comment (dethrace-labs) [reply src/DETHRACE/CMakeLists.txt:223]: How about set RUNTIME DESTINATION “.” as it was before maybe?

Comment (madebr) [reply src/DETHRACE/CMakeLists.txt:223]: That's fine!
If somebody ever packages this, he can move it himself :)

File: src/DETHRACE/CMakeLists.txt [hunk 2]
```diff
@@ -195,8 +202,40 @@ endif()
 
 if (DETHRACE_INSTALL)
     install(TARGETS dethrace
-        RUNTIME DESTINATION "."
+        BUNDLE DESTINATION . COMPONENT Runtime
+        RUNTIME DESTINATION . COMPONENT Runtime
     )
+
+    if (APPLE)
```
Comment (madebr) [inline src/DETHRACE/CMakeLists.txt:209]: Can you fix the indentation and make sure all cmake commands are lower case?

File: src/DETHRACE/CMakeLists.txt [hunk 3]
```diff
@@ -195,8 +200,38 @@ endif()
...
+            MACOSX_BUNDLE_BUNDLE_VERSION ${DETHRACE_VERSION}
+            MACOSX_BUNDLE_SHORT_VERSION_STRING ${DETHRACE_VERSION}
+            MACOSX_BUNDLE_ICON_FILE dethrace.icns
+        )
+        set_source_files_properties(${CMAKE_SOURCE_DIR}/packaging/macos/dethrace.icns PROPERTIES
+            MACOSX_PACKAGE_LOCATION "Resources"
+        )
+        target_sources(dethrace_obj PRIVATE
+            ${CMAKE_SOURCE_DIR}/packaging/macos/dethrace.icns
+        )
```
Comment (madebr) [inline src/DETHRACE/CMakeLists.txt:222]: Do we need an option to build DethRace on macos *not* as a bundle, as an ordinary cli executable?

Comment (dethrace-labs) [reply src/DETHRACE/CMakeLists.txt:222]: without `-DDETHRACE_INSTALL`, and ordinary executable is created

Comment (madebr) [reply src/DETHRACE/CMakeLists.txt:220]: True :)
That seems good enough for me.

**Pull request merged**